reporterdashboard/dashboard.py

```diff
@@ -0,0 +1,112 @@
+import json
+import argparse
+
+
+def get_matching_test_cases(test_cases_1, test_cases_2):
+    """Returns test cases that are both in test_cases_1 and test_cases_2.
+
+    Test cases are ordered by their name.
+    """
+    # create sorted copies
+    sorted_1 = sorted(test_cases_1, key=lambda d: d["name"])
+    sorted_2 = sorted(test_cases_2, key=lambda d: d["name"])
+    overlapping_test_cases = []
+
+    while True:
+        if len(sorted_1) == 0 or len(sorted_2) == 0:
+            break
+
+        if sorted_1[0]["name"] == sorted_2[0]["name"]:
+            # only include passing tests
+            if sorted_1[0]["result"] != "pass" or sorted_2[0]["result"] != "pass":
+                sorted_1.pop(0)
+                sorted_2.pop(0)
+                continue
+            overlapping_test_cases.append((sorted_1.pop(0), sorted_2.pop(0)))
+        elif sorted_1[0]["name"] < sorted_2[0]["name"]:
+            sorted_1.pop(0)
+        else:
+            sorted_2.pop(0)
+    return overlapping_test_cases
+
+
+def print_energy_differences(data):
```

Comment:
Whats the type here? `def print_energy_differences(data: list[tuple[?, ?]]):`

Reply:
Fixed